discovery/utils.go

```diff
@@ -0,0 +1,92 @@
+package discovery
+
+import (
+	"encoding/binary"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// waitingProofKey is the proof key which uniquely identifies the
+// announcement signature message. The goal of this key is distinguish the
+// local and remote proof fot the same channel id.
+// TODO(andrew.shvv) move to the channeldb package after waiting proof map
+// becomes persistent.
+type waitingProofKey [10]byte
```

Comment:
Why 10 bytes? Seems it can fit in `9` bytes. Better yet, this could itself just be a struct: 
```go
type waitingProofKey struct {
    chanID uint64
    remote bool
}
```
That would make for an even more compact in-memory representation.